### DISCUSSION TITLE
Mode change feedback flash

### DISCUSSION THREAD

**Discussion by alper** [nikitabobko/AeroSpace#970] [feature-ideas]
It would be nice if the major mode changes would flash a quick feedback toast so that you know what you're doing and that it's doing something.

These should be a setting so that users who are more advanced can then turn it off if they so like.

You could use that for config load errors as well (which right now pops up a terminal window).

**Comment by Coobaha** [comment]
I searched for a related feature and came across this issue. Here are my 2 cents:
- Add more callbacks to the configuration, such as `on-mode-change`.
- Enable the ability to query the current state of aerospace with `aerospace current mode`.

My use case was to run a script based on different modes and add borders to focused window only in some of the modes

**Comment by prgres** [comment]
Or add a field within the mode declaration that can run some script. Previously, I have done something similar with JankyBorders and skhd. After entering a mode, a message was sent to the borders app and the color of the borders change

**Comment by nemo-lm** [comment]
As an alternative to a flash, I'd prefer adding background-color to the menubar mode identifier.
